### PR TITLE
Created classes for parsing certificates, keys and secrets identifiers.

### PR DESCRIPTION
**Pull request by vcolin7**
Addresses #8292.

I'm not sure if the Identifier classes should belong to the `model` package, but they're there for now. Had to duplicate some code because we don't have a shared package for Key Vault in Java.

~~Will add unit tests soon.~~ Done.

### CODE REVIEW

**Comment by vcolin7** [comment]
> Should we write tests for these parsers?

@sadasant I saw some tests for these in .NET so I assumed so.

**Comment by sadasant** [comment]
@vcolin7 thank you! I'll ask you some last questions over team before I approve. Ping me if I don't ping you first.

**Comment by sadasant** [comment]
@vcolin7 thank you for reaching out to me! I believe the conversation helped us a lot.

**Review by sadasant — APPROVED** [review]
Thank you!

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/KeyVaultKeyIdentifier.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     * @return a new instance of {@link KeyVaultKeyIdentifier}.
+     * @throws IllegalArgumentException if the given identifier is {@code null}.
+     * @throws MalformedURLException if the given identifier is not a valid Key Vault Key identifier
+     */
+    public static KeyVaultKeyIdentifier parse(String keyId) throws IllegalArgumentException, MalformedURLException {
+        if (keyId == null) {
+            throw new IllegalArgumentException("keyId cannot be null");
+        }
+
+        URL url = new URL(keyId);
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/KeyVaultKeyIdentifier.java:84]: same feedback as certs.

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/KeyVaultKeyIdentifier.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        String[] pathSegments = url.getPath().split("/");
+
+        if ((pathSegments.length != 3 && pathSegments.length != 4) // More or less segments in the URI than expected.
+            || !"https".equals(url.getProtocol()) // Invalid protocol.
+            || (!"keys".equals(pathSegments[1]) && !"deletedkeys".equals(pathSegments[1])) // Invalid collection.
+            || ("deletedkeys".equals(pathSegments[1]) && pathSegments.length == 4)) { // Deleted items do not include a version.
+            throw new IllegalArgumentException("keyId is not a valid Key Vault Key identifier");
+        }
+
+        return new KeyVaultKeyIdentifier(keyId, url.getProtocol() + "://" + url.getHost(), pathSegments[2],
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/KeyVaultKeyIdentifier.java:95]: same feedback as certs.

File: sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/KeyVaultSecretIdentifier.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+     * @return a new instance of {@link KeyVaultSecretIdentifier}.
+     * @throws IllegalArgumentException if the given identifier is {@code null}.
+     * @throws MalformedURLException if the given identifier is not a valid Key Vault Secret identifier
+     */
+    public static KeyVaultSecretIdentifier parse(String secretId) throws IllegalArgumentException, MalformedURLException {
+        if (secretId == null) {
+            throw new IllegalArgumentException("secretId cannot be null");
+        }
+
+        URL url = new URL(secretId);
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/KeyVaultSecretIdentifier.java:84]: same feedback.

File: sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/KeyVaultSecretIdentifier.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        String[] pathSegments = url.getPath().split("/");
+
+        if ((pathSegments.length != 3 && pathSegments.length != 4) // More or less segments in the URI than expected.
+            || !"https".equals(url.getProtocol()) // Invalid protocol.
+            || (!"secrets".equals(pathSegments[1]) && !"deletedsecrets".equals(pathSegments[1])) // Invalid collection.
+            || ("deletedsecrets".equals(pathSegments[1]) && pathSegments.length == 4)) { // Deleted items do not include a version.
+            throw new IllegalArgumentException("secretId is not a valid Key Vault Secret identifier");
+        }
+
+        return new KeyVaultSecretIdentifier(secretId, url.getProtocol() + "://" + url.getHost(), pathSegments[2],
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/KeyVaultSecretIdentifier.java:95]: same feedback.

**Pull request merged**